doc/services/shell/index.rst

```diff
@@ -74,6 +74,28 @@ procedure:
   to the shell.
 
 
+Telnet Backend
```

Comment:
Suggestion: I'm pretty sure they're gonna want this section down after the current ["RTT Backend Channel Selection"](https://docs.zephyrproject.org/latest/services/shell/index.html#rtt-backend-channel-selection) section. I also think it would be helpful to link to the respective backend sections from the bulleted list above that already has "Segger RTT" and "Telnet".

Reply:
It's not an accident that I put it here instead of in the "RTT Backend Channel Selection". If you look above this section ([Connecting to Segger RTT via TCP (on macOS, for example)](https://github.com/DynonAvionics/zephyr/pull/4/files#diff-e428530c93b80c61b42c3fcabed8425c8e8d96297a48f16c70ec728d15aa4244L53)), I figured that it would be more logical to put this how-to-connect-to-telnet above the how-to-connect-to-rtt. I'm willing to take the risk of them asking for changes.